cmd/konvoy-image/cmd/validate.go

```diff
@@ -0,0 +1,73 @@
+package cmd
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/mesosphere/konvoy-image-builder/pkg/app"
+)
+
+const validEndpointSize = 2
+
+var validateFlags app.ValidateFlags
+
+// validateCmd runs validations against nodes to provision.
+var validateCmd = &cobra.Command{
+	SilenceUsage:  true,
+	SilenceErrors: true,
+	Use:           "validate",
+	Short:         "validate existing infrastructure",
+	Args:          cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if !isValidCIDR(validateFlags.PodSubnet) {
+			return fmt.Errorf("pod-subnet %q was not a valid CIDR", validateFlags.PodSubnet)
+		}
+
+		if !isValidCIDR(validateFlags.ServiceSubnet) {
+			return fmt.Errorf("service-subnet %q was not a valid CIDR", validateFlags.ServiceSubnet)
+		}
+
+		if err := validateEndpoint(validateFlags.APIServerEndpoint); err != nil {
+			return errors.Wrap(err, fmt.Sprintf("apiserver-endpoint %q was not a valid, ", validateFlags.APIServerEndpoint))
+		}
+
+		if err := app.Validate(args[0], validateFlags); err != nil {
+			return errors.Wrap(err, "error running provision")
+		}
+
+		return nil
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(validateCmd)
+
+	flagSet := validateCmd.Flags()
+	flagSet.StringVar(&validateFlags.ServiceSubnet, "service-subnet", "10.96.0.0/12", "ip addresses used"+
+		" for the service subnet")
+	flagSet.StringVar(&validateFlags.PodSubnet, "pod-subnet", "192.168.0.0/16", "ip addresses used"+
+		" for the pod subnet")
+	flagSet.StringVar(&validateFlags.APIServerEndpoint, "apiserver-endpoint", "", "required - apiserver endpoint")
```

Comment:
Lets make this optional please. (Probably means changing some Ansible)

Reply:
You said yesterday this should fail if not defined.

Reply:
Sorry I don't remember that, but if I did I misspoke.
The person running these checks might not have this information.

Reply:
Maybe this can be completely removed. The only check is for the port, and that port `6443` can be configured to be something else.

Reply:
I will modify it to just have the apiserver Endpoint port, check separately for that port in another task if apiserver_port is defined while have the main task go through the necessary static ports we are checking for